api/app/Generators/CandidateProfileCsv.php

```diff
@@ -82,132 +115,127 @@ public function generate()
             ],
         ])
             ->whereIn('id', $this->ids)
-            ->get();
-
-        if (empty($candidates)) {
-            throw new Exception('No candidates found.');
-        }
-
-        $sheet = $this->spreadsheet->getActiveSheet();
+            // This is being very weird with Laratrust
+            //->authorizedToView(['userId' => $this->userId])
+            ->chunk(200, function ($candidates) use ($sheet, &$currentCandidate) {
+                foreach ($candidates as $candidate) {
+
+                    $department = $candidate->user->department()->first();
+                    $preferences = $candidate->user->getOperationalRequirements();
+                    $educationRequirementExperiences = $candidate->educationRequirementExperiences->map(function ($experience) {
+                        return $experience->getTitle();
+                    })->flatten()->unique()->toArray();
+
+                    $values = [
+                        $this->localizeEnum($candidate->pool_candidate_status, PoolCandidateStatus::class), // Status
+                        $this->localizeEnum($candidate->user->priority, PriorityWeight::class),
+                        '', // Availability
+                        $this->sanitizeString($candidate->notes ?? ''), // Notes
+                        $this->localizeEnum($candidate->user->current_province, ProvinceOrTerritory::class), // Current province
+                        $candidate->submitted_at ? $candidate->submitted_at->format('Y-m-d') : '', // Date received
+                        $candidate->expiry_date ? $candidate->expiry_date->format('Y-m-d') : '', // Expiry date
+                        $candidate->archived_at ? $candidate->archival_at->format('Y-m-d') : '', // Archival date
```

Comment:
Is it worth abstracting out to a format helper? :thinking:

Reply:
Maybe if the same format is used in other files? Not if its just these three lines.

Reply:
Yeah, I feel like formatting dates will be used but I'm not sure about the format so I tihnk for now it seems fine to stay here.

If we notice we continually format them in the same way in other files we could probably move it to the base `FileGenerator` class :+1: